src/oidcprovider/adapters/utils.ts

```diff
@@ -63,3 +63,23 @@ export const makeNotImplementedAdapter = (
     return promiseRejected();
   },
 });
+
+/**
+ * Given a date return the number of seconds from 1970-01-01T00:00:00Z UTC
+ * until the specified UTC date/time, ignoring leap seconds.
+ */
+export const toNumericDate = (date: Date): number => date.getTime() / 1000;
+/**
+ * Given a NumericDate returns a Date.
+ * NumericDate is a numberic value representing the number of seconds from
+ * 1970-01-01T00:00:00Z UTC until the specified UTC date/time,
+ * ignoring leap seconds. This is equivalent to the IEEE Std 1003.1,
+ * 2013 Edition [POSIX.1] definition "Seconds Since the Epoch", in
+ * which each day is accounted for by exactly 86400 seconds, other
+ * than that non-integer values can be represented.
+ */
+export const fromNumericDate = (numericDate: number): Date => {
+  const date = new Date(0);
+  date.setUTCSeconds(numericDate);
+  return date;
+};
```

Comment:
```suggestion
export const fromNumericDate = (numericDate: number): Date => {
  return new Date(numericDate * 1000);
};
```
just an alternative that works too.

Anyway your implementation works too, I tested it with this small test:
```
describe("fromNumericDate", () => {
  it("should create a correct date", () => {
    fc.assert(
      fc.property(fc.nat(), (seconds) => {
        const actual = fromNumericDate(seconds);
        const expected = new Date(seconds * 1000);
        expect(actual).toStrictEqual(expected);
      })
    )
  })
});
```

I had some doubts because of what the documentation [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setUTCSeconds) said.

Reply:
Much better!